solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+
+                // update network conversion amounts
+                networkTokenConversionAmounts[i] = networkTokenConversionAmount;
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(networkTokenConversionAmount);
+            }
+        }
+
+        // calculate the burn incentive fee and reduce it from the total amount to convert
+        (uint256 netNetworkTokenConversionAmount, uint256 incentiveFeeAmount) =
+            applyIncentiveFee(grossNetworkTokenConversionAmount);
+
+        // approve the governance token converter to withdraw the network token amount
+        ensureAllowance(_networkToken, network, netNetworkTokenConversionAmount);
+
+        // convert all network token amounts to the governance token
+        totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(
+            network.convertByPath2(govPath, netNetworkTokenConversionAmount, 1, address(this))
+        );
+
+        // update the stats of the vortex
+        _totalBurnedAmount = _totalBurnedAmount.add(totalGovTokenAmountToBurn);
+
+        // burn all the converter governance tokens
+        _govTokenGovernance.burn(totalGovTokenAmountToBurn);
+
+        // transfer the incentive fee to the caller
+        _networkToken.transfer(msg.sender, incentiveFeeAmount);
+
+        emit Burned(tokens, amounts, networkTokenConversionAmounts, totalGovTokenAmountToBurn);
+    }
+
+    /**
+     * @dev returns the result of the conversion the provided tokens to the governance token and its burn
+     *
+     * @param tokens the tokens to convert
+     *
+     * @return the amounts of the converted tokens
+     * @return the network token amounts the tokens were converted to
+     * @return the total burned amount in this vortex
+     * @return the incentive fee resulting from this vortex
+     */
+    function availableVortex(IERC20[] calldata tokens)
+        external
+        view
+        returns (
+            uint256[] memory,
+            uint256[] memory,
+            uint256,
+            uint256
+        )
+    {
+        // check for duplicates in order to behave similarly to the vortex function
+        require(!hasDuplicates(tokens), "ERR_INVALID_TOKEN_LIST");
+
+        // retrieve conversion paths and the amounts to burn
+        (address[][] memory paths, uint256[] memory amounts, address[] memory govPath) =
+            vortexStrategy(tokens, networkFeeWallet());
+
+        // get all network token conversion amounts
+        BancorNetwork network = bancorNetwork();
+
+        uint256[] memory networkTokenConversionAmounts = new uint256[](tokens.length);
+        uint256 grossNetworkTokenConversionAmount = 0;
+        uint256 totalGovTokenAmountToBurn = 0;
+
+        for (uint256 i = 0; i < paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = amounts[i];
+            if (amount == 0) {
+                continue;
```

Comment:
Same comment as previous (why not filter out these amounts in `vortexStrategy`?).

Reply:
Fixed